runtime/compiler/runtime/JITServerAOTCache.cpp

```diff
@@ -58,13 +64,46 @@ AOTCacheClassLoaderRecord::AOTCacheClassLoaderRecord(uintptr_t id, const uint8_t
    {
    }
 
+AOTCacheClassLoaderRecord::AOTCacheClassLoaderRecord() :
+   _data()
+   {
+   }
+
 AOTCacheClassLoaderRecord *
 AOTCacheClassLoaderRecord::create(uintptr_t id, const uint8_t *name, size_t nameLength)
    {
    void *ptr = AOTCacheRecord::allocate(size(nameLength));
    return new (ptr) AOTCacheClassLoaderRecord(id, name, nameLength);
    }
 
+AOTCacheClassLoaderRecord *
+AOTCacheClassLoaderRecord::create(size_t nameLength)
+   {
+   void *ptr = AOTCacheRecord::allocate(size(nameLength));
+   return new (ptr) AOTCacheClassLoaderRecord();
+   }
+
+AOTCacheClassLoaderRecord *
+AOTCacheClassLoaderRecord::read(FILE *f)
+   {
+   ClassLoaderSerializationRecord header;
+   if (1 != fread(&header, sizeof(header), 1, f))
+      return NULL;
+
+   if ((AOTSerializationRecordType::ClassLoader != header.type()) ||
+       (0 == header.id()))
+      return NULL;
+
+   AOTCacheClassLoaderRecord *record = create(header._nameLength);
+   memcpy((void *)record->dataAddr(), &header, sizeof(header));
+   if (1 != fread((void *)record->data()._name, header._nameLength, 1, f))
```

Comment:
This can be unified across record types as reading `header.size() - sizeof(header)` bytes into `(uint8_t *)record->dataAddr() + sizeof(header)`.